### PR TITLE
DDF-4424 Fixing CDM issues with GSON deserialization and ingesting when the local catalog is unavailable

### PR DESCRIPTION
**Pull request by Kjames5269**
Uncomment this block and include PR # of change against 2.13.x where appropriate.
When doing so, it is reasonable to delete much of the rest of the PR description, leaving only:
- This uncommented block with link to 2.13.x PR
- Reviewers tagged
- Teams tagged
- Committers tagged
- The link to the "Notes on Review Process"

##### ABBREVIATED REVIEW BETWEEN 2.13.X AND MASTER IS IN EFFECT
Link to 2.13.x PRs: #4630 #4617
____

#### Who is reviewing it? 
@clockard @peterhuffer @kcover @lessarderic @brjeter @jhunzik 

#### Select relevant component teams: 
@codice/io 

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by Kjames5269** [comment]
build now

File: catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/FileTypeAdapter.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ *
+ * <p>https://github.com/google/gson/issues/414#issuecomment-237377182
+ */
+public class FileTypeAdapter implements JsonDeserializer<File> {
+  @Override
+  public File deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context)
+      throws JsonParseException {
+    return new File(((JsonObject) json).get("path").getAsString());
+  }
+}
```
Comment (bakejeyner) [inline catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/FileTypeAdapter.java]: ❓ Is this the main fix?

**Comment by Kjames5269** [comment]
build now

**Comment by peterhuffer** [comment]
Successful hero ✅ 
Verified ingesting with in-place directory monitor. Turned off DDF. Created, updated, and deleted files in the monitored directory. Turned DDF back on and verified the catalog was correct.

File: catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/JsonPersistantStore.java [hunk 2]
```diff
@@ -37,7 +39,10 @@
 
   private static final String PERSISTED_FILE_SUFFIX = ".json";
 
-  private Gson gson = new Gson();
+  private Gson gson =
+      new GsonBuilder()
+          .registerTypeAdapter(new TypeToken<File>() {}.getType(), new FileTypeAdapter())
+          .create();
```
[the remark this answers is not in the document]
Comment (Kjames5269) [reply catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/JsonPersistantStore.java]: Yes and no. It's one of the two, The other is [here](https://github.com/codice/ddf/pull/4637/files#diff-5a5305f3c20ea015d7e17b46947ef8ddR150) to fix an issue with reingesting when the store isn't available

**Pull request merged**